Support for empty CANNOTIMPORT

mweb opened the pull request:
With this pull request it is possible to specify an IMPORTRULE without a CANNOTIMPORT if an ALLOW is specified. This is useful if the default policy of a project is to prohibit all kind of imports and that each new import needs to be approved. Thiis also helps if a package gets renamed and the .importrules file was forgotton to be updated.

Commit:
add documentation and example for block every import with exception

Commit:
use ALLOW within .importrules of the project

lucianonooijen commented:
Thanks a lot for creating this merge request. It looks like a good addition to the tool. We should also keep in mind that implementing this would mean you're also blocked from using the standard library. In my opinion a good thing, although it should be clarified.

For this pull request, I think it's best to create a new keyword, like `ONLYALLOW`, so there are two options when defining import rules:
- `IMPORTRULE [pkg] CANNOTIMPORT [pkg] ALLOW [pkg];`, where `ALLOW [pkg]` is optional
- `IMPORTRULE [pkg] ONLYALLOW [pkg];`

For accepting this pull request, could you remove the one-line change in `parser/parser.go` ([link](https://github.com/BytecodeAgency/import-boundary-checker/pull/4/files#diff-a516a27d5e0a1280de56136d7c3d5d36R133)). Of course implementing the keyword is fine too, but don't feel forced to do it 🙂.

mweb commented:
I might not have been specific enough this still allows to use the all other packages. The goal is to prohibit all the imports from my own package to enforce the architecture without specifying everything that I might not allow.
Since the import rules can be extended I would use CANNOTIMPORT for the howl application and then add special rules within the sub packages.

`IMPORTRULE [pkg] CANNOTIMPORT [PKG];`
`IMPORTRULE [pkg]/usecase ALLOW [pkg]/domain;`
`IMPORTRULE [pkg]/drivers ALLOW "[pkg]/domain" "[pkg]/usecase";`


With the one-line change I am allowed to write the second IMPORTRULE, otherwise the parser complains that the CANNOTIMPORT is empty.

lucianonooijen commented on examples/go-invalid-5/README.md:
```diff
@@ -0,0 +1,6 @@
+
+All packages are not allowed to import any package from within the project. The
+data package is allowed import anything from within the data package and their
+subpackages. Teh domain/user package imports from data/interactions which is not
```
Typo: `Teh`

lucianonooijen reviewed (changes requested):
Ah I see, I misunderstood the purpose. Could you clarify this a bit more in the docs (like the example you gave in the comment here)? I also found a little typo. After that it can me merged into master 😄

mweb commented:
Ok I fixed the comment and changed the example I've added to be similar to the one I provided in the comments here. Is this what you wanted? Or should I add an example of a nested project config within the README.md?

lucianonooijen commented:
> Or should I add an example of a nested project config within the README.md? 

That would be a great addition! PR looks great so when the README is updated, it can be merged 😄

Commit:
add description about block all configuration
